echopype/preprocess/api.py

```diff
@@ -64,18 +64,32 @@ def _freq_MVBS(ds, rint, pbin):
     range_interval = np.arange(
         0, ds_Sv["echo_range"].max() + range_meter_bin, range_meter_bin
     )
-    MVBS = ds_Sv.groupby("frequency").apply(
-        _freq_MVBS, args=(range_interval, ping_time_bin)
+    ds_MVBS = (
+        ds_Sv.groupby("frequency")
+        .apply(_freq_MVBS, args=(range_interval, ping_time_bin))
+        .to_dataset()
     )
-
     # Attach attributes
-    MVBS.attrs = {
+    ds_MVBS["ping_time"].attrs = {
+        "long_name": "Timestamp of each ping",
```

Comment:
@leewujung this MVBS `ping_time` long_name doesn't seem quite right, given the MVBS gridding. But I couldn't come up with something better. Plus, is keeping `ping_time` as the time dimension name even appropriate given the gridding?

Reply:
Hmm... I think it should still be `ping_time` since conceptually is still the same thing as before the averaging operation. But you are right, the long_name should be change to something else that reflects what it really is.

"Leading timestamp of the averaging time window"  ??

Reply:
> Hmm... I think it should still be ping_time since conceptually is still the same thing as before the averaging operation.

Yeah. I guess I was getting hung up on the fact that with MVBS we're not dealing with individual pings any more.

> "Leading timestamp of the averaging time window" ??

It's getting there, and it looks factually correct. But, also, yuck :laughing:. Maybe we should take a look at the [CF concept of "cells"](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#_data_representative_of_cells) and [cell "methods"](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#cell-methods).

Come to think of it, the same logic would apply to `range` (stating that it's based on an averaging window).

Reply:
Agree with corresponding changes to `range`.  In a way this has the "provenance" in the dataset also. But of course we will need something actually added to the `Provenance` group (#221 ).

Reply:
I've reviewed the CF [cell methods](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#cell-methods) section, including the sub-sections [Statistics for more than one axis](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#statistics-more-than-one-axis) and [Recording the spacing of the original data and other information](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#recording-spacing-original-data). I'll start with a question / suggestion: `ping_time` is really just a timestamp, isn't it? The original clock (the source of the timestamps) is of course the transducer's clock, but for MVBS we're dealing with gridded/binned Sv and the time dimension can be generalized to simply `time`, with a generic `long_name` ("time"?). Do you see a problem with doing that?

Then, for the MVBS variable (`Sv`), I recommend adding some iteration of the following attribute, where in this example time binning is 20 sec and range binning is 10 m:
```
cell_methods = "time: mean (interval: 20 second comment: time is the interval start) echo_range: mean (interval: 20 meter comment: echo_range is the interval start)"
```

CF specifies that when using `cell_methods`, cell bound on the corresponding dimensions should also be included as ancillary variables. For expedience (we have a lot of other work to do for 0.6.0!) I suggest we skip that effort right now, and work on adding those variables in the future.

Reply:
I've implemented the recommendations I described above, so you can assess them more easily. Here's a screenshot of a resulting MVBS dataset:

![image](https://user-images.githubusercontent.com/742403/161011947-d4ab50cf-37de-43f2-8fbc-3fa3ae91905a.png)

Reply:
Thanks @emiliom! I agree with the additions, and that it is nice that the `cell_methods` give very concise and precise description. Could you create an issue to make sure that we add in the cell bounds?

I think though we should keep `ping_time` instead of changing that to `time`. It is more intuitive for acoustics people, explicit that this MVBS is just a mean of Sv so has the same coordinate, and would avoid confusion with time1, time2, etc in the other groups.

Reply:
> I think though we should keep ping_time instead of changing that to time

Will do

Reply:
@leewujung  I've reverted the time coord name to `ping_time`. All tests ran successfully, locally.

Note: once we're good with this, it looks like we should also add attributes to `compute_MVBS_index_binning`. I've never used this function before, so I wanted to hold off until we're in full agreement.

Reply:
Sigh, the tests are failing again. I think I finally see what's going on. I'll get to it

Reply:
Sounds good. Do you want to handle adding attribute to `compute_MVBS_index_binning ` in this PR? It seems better to have that in a separate PR so we can get this fully tested first.

Reply:
Tackling compute_MVBS_index_binning attributes in a separate PR sounds good. I'm all for completing this PR soon!